docs/specs.html

```diff
@@ -1253,12 +1254,225 @@ <h4 id="__span_class_scenario_name_equals_span"><span class="scenario name">Equa
+
+Scenario: Object in response body is only a valid Object.
+When the response body is assigned:
+"""
+{"foo": 1}
+"""
+Then the value of the response body is not a valid `String`
+And the value of the response body is not a valid `Number`
+And the value of the response body is not a valid `Integer`
+And the value of the response body is a valid `Object`
+And the value of the response body is not a valid `Array`
+And the value of the response body is not a valid `Boolean`
+
+Scenario: Quoted Object in response body is only a valid String.
+When the response body is assigned:
+"""
+"{"foo": 1}"
+"""
+Then the value of the response body is a valid `String`
+And the value of the response body is not a valid `Number`
+And the value of the response body is not a valid `Integer`
+And the value of the response body is not a valid `Object`
+And the value of the response body is not a valid `Array`
+And the value of the response body is not a valid `Boolean`
+
+
+Scenario: Empty Array in response body is only a valid Array.
+When the response body is assigned:
+"""
+[]
+"""
+Then the value of the response body is not a valid `String`
+And the value of the response body is not a valid `Number`
+And the value of the response body is not a valid `Integer`
+And the value of the response body is not a valid `Object`
+And the value of the response body is a valid `Array`
+And the value of the response body is not a valid `Boolean`
+
+Scenario: Array in response body is only a valid Array.
+When the response body is assigned:
+"""
+[1, "foo"]
+"""
+Then the value of the response body is not a valid `String`
+And the value of the response body is not a valid `Number`
+And the value of the response body is not a valid `Integer`
+And the value of the response body is not a valid `Object`
+And the value of the response body is a valid `Array`
+And the value of the response body is not a valid `Boolean`
+
+Scenario: true in response body is only a valid Boolean.
+When the response body is assigned:
+"""
+true
+"""
+Then the value of the response body is not a valid `String`
+And the value of the response body is not a valid `Number`
+And the value of the response body is not a valid `Integer`
+And the value of the response body is not a valid `Object`
+And the value of the response body is not a valid `Array`
+And the value of the response body is a valid `Boolean`
+
+Scenario: false in response body is only a valid Boolean.
+When the response body is assigned:
+"""
+false
+"""
+Then the value of the response body is not a valid `String`
+And the value of the response body is not a valid `Number`
+And the value of the response body is not a valid `Integer`
+And the value of the response body is not a valid `Object`
+And the value of the response body is not a valid `Array`
+And the value of the response body is a valid `Boolean`
+
+Scenario: null in response body is not any valid type.
+When the response body is assigned:
+"""
+[null]
```

Comment:
this isn't an array?

Reply:
That is, but the tests access the first element. That may have been done to use the JSON parser since otherwise `null` will need an argument transform.